var/spack/repos/builtin/packages/slate/package.py

```diff
@@ -54,3 +56,27 @@ def cmake_args(self):
             '-Duse_mpi=%s'           % ('+mpi' in spec),
             '-DSCALAPACK_LIBRARIES=%s'    % spec['scalapack'].libs.joined(';')
         ]
+
+    examples_src_dir = 'examples'
+
+    @run_after('install')
+    def cache_test_sources(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        if self.spec.satisfies('@master'):
+            self.cache_extra_test_sources([self.examples_src_dir])
+
+    def test(self):
+        if self.spec.satisfies('@master') and '+mpi' in self.spec:
```

Comment:
Please output that the tests are being skipped if these criteria are *not* met to help with debugging.  

Perhaps print something like "Skipping: Stand-alone tests only run on master with +mpi" and return.  Then you can move the remaining lines in 4 spaces (so more readable and consistent with style preferences).